src/main/java/pl/psnc/dei/ui/components/batches/DatasetImportComponent.java

```diff
@@ -0,0 +1,308 @@
+        Button button = new Button("Upload records");
+        button.addClickListener(event -> uploadRecords());
+        return button;
+    }
+
+    private Checkbox prepareCreateImportCheckbox() {
+        createImportCheckbox = new Checkbox();
+        createImportCheckbox.setLabel("Create import");
+        createImportCheckbox.setValue(doCreateImport);
+        createImportCheckbox.addClassName("mt-16px");
+        createImportCheckbox.addValueChangeListener(event -> {
+            doCreateImport = event.getValue();
+            importNameTextField.setEnabled(doCreateImport);
+            splitRecordsCheckboxWithNumber.setEnabled(doCreateImport);
+        });
+        return createImportCheckbox;
+    }
+
+    private Component createAggregatorSelection() {
+        Predicate<Aggregator> predicate = agg -> agg.equals(Aggregator.EUROPEANA);
+        aggregatorsSelect = CommonComponentsFactory.getAggregatorSelector(predicate);
+        aggregatorsSelect.setEnabled(false);
+        aggregatorsSelect.setValue(Aggregator.EUROPEANA);
+        return aggregatorsSelect;
+    }
+
+    private Component createProjectSelection() {
+        ListDataProvider<Project> projectsProvider = new ListDataProvider<>(projectsRepository.findAll());
+        projectsSelect = CommonComponentsFactory.getProjectSelector(projectsProvider);
+        projectsSelect.addValueChangeListener(event -> {
+            project = event.getValue();
+            datasetsSelect.setItems(batchService.getProjectDataset(project));
+        });
+        return projectsSelect;
+    }
+
+    private Component createDatasetsSelection() {
+        datasetsSelect = CommonComponentsFactory.getDatasetSelector(Collections.emptyList());
+        datasetsSelect.addValueChangeListener(event -> dataset = event.getValue());
+        datasetsSelect.setEmptySelectionAllowed(true);
+        return datasetsSelect;
+    }
+
+    private TextArea prepareExcludedTextArea() {
+        excludeSelectedTextArea = new TextArea();
+        excludeSelectedTextArea.setMinHeight("150px");
+        excludeSelectedTextArea.addClassName("flex-1");
+        excludeSelectedTextArea.setHelperText("Enter each Europeana ID in separate line or put JSON structured array of record IDs.");
+        excludeSelectedTextArea.setEnabled(doExcludeSelected);
+        return excludeSelectedTextArea;
+    }
+
+    private AccordionPanel preparePreviewAccordion() {
+        previewAccordion = new AccordionPanel(getEmptyDatasetPreviewHeader(), null);
+        previewAccordion.setEnabled(false);
+        return previewAccordion;
+    }
+
+    private AccordionPanel fillAccordion(String datasetName, int totalRows, List<RecordSample> recordSamples) {
+        previewAccordion.setSummaryText(getDatasetPreviewHeader(datasetName, totalRows));
+        previewAccordion.setContent(preparePreviewRecordsGrid(recordSamples));
+        previewAccordion.setEnabled(true);
+        return previewAccordion;
```

Comment:
I am not sure if you will use return value, but so far it is dropped after method execution

Reply:
Yeah, this could be void

Reply:
I mean, there are task ongoing, which are related to this code, so I am not sure if it is intended to return value, because some method will use it, or it is remnant of some changes